01_Data/src/layers/sequelize/repository/TransactionEvent.ts

```diff
@@ -143,11 +155,74 @@ export class SequelizeTransactionEventRepository extends SequelizeRepository<Tra
     });
   }
 
+  private async calculateAndUpdateTotalKwh(transaction: Transaction): Promise<number> {
+    const totalKwh = this.getTotalKwh(transaction.meterValues ?? []);
+
+    await transaction.update(
+      {totalKwh}
+    );
+
+    return totalKwh;
+  }
+
+  private getTotalKwh(meterValues: MeterValueType[]): number {
+    const contexts: ReadingContextEnumType[] = [
+      ReadingContextEnumType.Transaction_Begin,
+      ReadingContextEnumType.Sample_Periodic,
```

Comment:
technically, you don't need Sample_Periodic, only begin and end. will make the rest of this easier.